pkg/types/baremetal/platform.go

```diff
@@ -6,9 +6,10 @@ import (
 
 // BMC stores the information about a baremetal host's management controller.
 type BMC struct {
-	Username string `json:"username"`
-	Password string `json:"password"`
-	Address  string `json:"address"`
+	Username                       string `json:"username"`
+	Password                       string `json:"password"`
+	Address                        string `json:"address"`
+	DisableCertificateVerification bool   `json:"disableCertificateVerification"`
```

Comment:
I don't think this is a good security position to allow disabling verification. the openstack team worked to allows users to specify the certificate for the servers to that they can communicate correctly with the servers.

Reply:
It’s on our radar to support additional options but disabling has to be possible.  Especially for lab environments or development.  We could introduce an explicit warning but the requirement to disable verification isn’t going to go away.

Whatever openstack doing is not the same as bare metal talking to baseboard management controllers. It’s not as simple as trusting a specific cert.  We’re talking to these devices largely on IP’s, which aren’t even going to be part of the CN or SAN, they all ship with default self-signed certs that usually have some UUID in the name.

Reply:
Filed https://github.com/metal3-io/baremetal-operator/issues/420 to track this upstream